src/components/forms/CommentAdder.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { useContext, useState } from "react";
+import { UserContext } from "../../context/UserContext";
+import { addNewComment } from "../../utils/api";
+
+export default function CommentAdder({
+  article_id,
+  articleCommentList,
+  setArticleCommentList,
+}) {
+  const { users } = useContext(UserContext);
+  const [comment, setComment] = useState("");
+
+  const newComment = { username: users[0].username, body: comment };
+
+  const submitHandler = (e) => {
+    e.preventDefault();
+    addNewComment(newComment, article_id);
+    setArticleCommentList([comment, ...articleCommentList]);
```

Comment:
Be careful using optimistic rendering here - we currently get an error about a unique key prop.
We MUST wait for the new comment to be returned from the api as it has all the data we may want for a comment.
This also makes it tricky if the request fails!

When setting state, if you care about what state was previously you MUST use a function, otherwise we risk retaining access to stale data.

`setArticleComments((currentComments) => [comment, ...currentComments])`